central/detection/service/service_impl.go

```diff
@@ -140,6 +143,20 @@ func (s *serviceImpl) DetectBuildTime(ctx context.Context, req *apiV1.BuildDetec
 	if err != nil {
 		return nil, err
 	}
+
+	if req.GetCluster() != "" {
+		// The request indicates enrichment should be delegated to a specific cluster.
+		clusterID, err := clusterutil.GetClusterIDFromNameOrID(ctx, s.clusters, req.GetCluster())
+		if err != nil {
+			if errors.Is(err, clusterutil.ErrClusterNotFound) {
+				return nil, status.Error(codes.NotFound, "cluster not found, ensure cluster exists and have access")
```

Comment:
one point when using the errors from `roxerror` package: they will be automatically converted to the gRPC code, so no need to duplicate this logic here if we use them.

(if you are interested, here's [the related interceptor](https://github.com/stackrox/stackrox/blob/master/pkg/grpc/errors/interceptor.go#L95) that takes care of that)

Reply:
Done - refer to: https://github.com/stackrox/stackrox/pull/7738/commits/177eeddbf27caa0c7f50eb031220982b05b04fe3